### ISSUE TITLE
Post Request Stream=True

### ISSUE THREAD

**Issue by aldengolab** [cryzed/Selenium-Requests#8]
Thanks for this encapsulation, it's been extremely helpful. I encountered a problem when using the syntax `driver.request('POST', dl_url, stream = True)`: the returned response object does not produce any content despite receiving a 202 success and despite the headers containing every sign of success (`{'Content-Disposition': 'attachment; filename=Report.csv', 'Transfer-Encoding': 'chunked' ...`).

`stream` is a necessary option for POST downloads of chunk-encoded content in the requests library. Based on my understanding of your code, line 211 in `request.py` seems to leave out the stream option entirely, thus hard-coding in `stream=False`, as this is the default. Is that reading correct?

**Comment by cryzed** [comment]
> Is that reading correct?

I don't think so, no. If you check the [method-signature](https://github.com/cryzed/Selenium-Requests/blob/master/seleniumrequests/request.py#L134) you will see that it accepts `**kwargs` -- so any keyword-argument that isn't `find_window_handle_timeout` or `page_load_timeout` will be contained in the `kwargs` dictionary. This dictionary is passed later to to the Selenium session `request` method, i.e. the stream keyword-argument should definitely arrive there.

Can you possibly try to do whatever you did to get access to the `dl_url` in your code, and then instead of using Selenium-Requests use a requests session directly and see if it behaves any different? Are you using something like [iter_content](http://docs.python-requests.org/en/master/api/#requests.Response.iter_content) to read the response chunk by chunk?

**Comment by aldengolab** [comment]
Hm, alright, I appreciate the assistance. I'm having difficulty getting the headers to operate correctly with this particular site I'm scraping (thus why I was using your code here) but looks like I'll just have to give them another shot. 

Yes, I'm using `iter_content` to read the response, but the content is empty. Initiating the submit click for the POST call on the site interactively using Selenium yields a download prompt (I may wind up needing to just deal with that, I haven't cracked it yet), but a direct POST is having some sort of disconnect. With the HTTP Requests package, it _should_ just be a matter of adding `stream=True` then iterating over the generator for streamed responses. It could be something to do with the server set-up, in which case I'm going to have to find another work-around. 

Will update if I find this has anything to do with this package, but I'm doubtful it does. Thanks!

**Comment by cryzed** [comment]
>  The request has been accepted for processing, but the processing has not been completed. The request might or might not eventually be acted upon, as it might be disallowed when processing actually takes place. There is no facility for re-sending a status code from an asynchronous operation such as this.

> The 202 response is intentionally non-committal. Its purpose is to allow a server to accept a request for some other process (perhaps a batch-oriented process that is only run once per day) without requiring that the user agent's connection to the server persist until the process is completed. The entity returned with this response SHOULD include an indication of the request's current status and either a pointer to a status monitor or some estimate of when the user can expect the request to be fulfilled. 
[Source](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html)

So it seems to me that you are getting the 202 status code, because the request was successful but no data is yet available. It might be that the response body doesn't actually carry the target content, but some other resources that is indicated by the headers. Maybe there's some kind of redirect URL in the headers or some JavaScript on the site uses a value in the headers to start the actual download. If everything else fails try using the web developer network tools in your browser, which allows you to see exactly what happens HTTP-wise. If this doesn't quite work as expected, you can also try using [Charles](https://www.charlesproxy.com/).

**Comment by aldengolab** [comment]
I have now replicated the same issue using HTTP Requests rather than Selenium-Requests.

Initially, this did trip me up: I discovered that the XMLhttprequest object had a `onreadystatechange` function which used the initial request response to update a page attribute and then submit a hidden form; this form makes a new HTTP call to the server with a `_blank` target. This is `dl_url`. I am able to mimic this function call in my Selenium code (or just call it) producing a download dialog. However, upon making the POST call to `dl_url`, there is no content, despite successful connection.

Double checking the response code, it's actually 200, not 202, which indicates a closed connection. Headers here aren't particularly helpful:

`{'Content-Disposition': 'attachment; filename=Report.csv', 'Transfer-Encoding': 'chunked', 'Server': 'Apache/2.2.21 (CentOS)', 'Connection': 'close', 'Date': 'Wed, 22 Jun 2016 15:37:02 GMT', 'Content-Type': 'application/x-csv;charset=UTF-8'}`

It seems to me the issue is that the connection closes when it should remain open for streaming the chunked content. I wonder if Firefox keeps the connection open with the native dialog, but it closes with the POST call from python? I have considered re-writing my code to use some kind of XMLHttpRequest encapsulation like scrapy, but it doesn't seem necessary since these are asynchronous.
